### DISCUSSION TITLE
Wait for all child components to reply (via events)

### DISCUSSION THREAD

**Discussion by bitcanon** [livewire/livewire#4671] [1. Help] [answered]
Hi, I'm building an application that is made up of Livewire components at the moment.

It's layed out like this:
1. I have a few pages containing a subset of components (which gather input from the user)
2. On each page there is one main component (called ConfigBuilder) that emit a signal to all other components on the page.
3. When a component receives the signal, it instantly replies by emitting a return signal to the ConfigBuilder component.

The potential issue I'm seeing now is that maybe not all child components will be able to reply in time.

Is there a way for me to do it like this:
1. Main component tell child components to send their input data.
2. Child components send their data.
3. Main component wait until all data has arrived.
4. Main component then use all input data gathered from the child components to build the final data.

Any help or input would be most appreceated :)

**Comment by joshhanley** [comment, marked as answer]
@bitcanon I probably wouldn't structure it this way due to the issue of needing events to communicate between components. Personally I would just have the one main component and make everything else blade components instead of Livewire components.

Hope this helps!

**Reply by bitcanon** [reply]
@joshhanley Thanks for your input. The reason for me using Livewire components was to get the nice live validation. Is it possible to use a set of blade components and have the main Livewire component handle the validation of all the blade components?

**Comment by joshhanley** [comment]
@bitcanon yeah you can 🙂  it's how I write all my form components now!

**Reply by bitcanon** [reply]
@joshhanley That's great news! Although I don't understand how I would go about and do that. You don't happen to have an open source project where you're using this technique so I can have a look? Or some kind of example code?

**Comment by joshhanley** [comment]
@bitcanon I don't have anything public.

But essentially you can just make a blade component say for a text input

`resources\views\components\input.blade.php`
```blade
<div>
    <input type="text" {{ $attributes->class('put some css classes here to style your input') }} />
</div>
```

Then in your Livewire component just make use of your input component and add `wire:model` to it

`resources\views\livewire\my-component.blade.php`
```blade
<div>
    <x-input wire:model="myProperty" />
</div>
```

Hope this helps!

**Reply by bitcanon** [reply]
Ah, that makes perfect sense! Thanks alot for your help and have a great day! :)

**Reply by joshhanley** [reply]
No worries!
